tools/pkg/engine_build_configs/lib/src/build_config_runner.dart

```diff
@@ -579,6 +579,27 @@ final class BuildRunner extends Runner {
     return min(multiplier * cores, 1000);
   }();
 
+  static final RegExp _gccRegex =
+      RegExp(r'^(.+)(:\d+:\d+:\s+(?:error|note|warning):\s+.*)$');
+
+  /// Converts relative [path], who is relative to [dirPath] to a relative path
+  /// of the `CWD`.
+  static String _makeRelative(String path, String dirPath) {
+    final String abs = p.join(dirPath, path);
+    return './${p.relative(abs)}';
+  }
+
+  /// Takes a [line] from compilation and makes the path relative to `CWD` where
+  /// the paths are relative to [outDir].
+  static String fixGccPaths(String line, String outDir) {
```

Comment:
Should this be `@visibleForTesting` if that is the intent?

Reply:
I omit that when the thing that is made visible is a pure function.  Its visibility and usage can't hurt anything.  I reserve `@VisibleForTesting` for things I want to be private but can't because of testing and using them in production can invalidate state.  Happy to change it if that doesn't make sense.

Reply:
It's less of a "can't hurt anything" and more "uh oh, we're now using this in other parts of the repo unexpectedly".

Let's make this `@visibleForTesting` the same way we'd make the function private if we could (this is package `build_engine_config`, not package `gcc_stdio`), if that makes sense. LGTM assuming we do that.

Reply:
done